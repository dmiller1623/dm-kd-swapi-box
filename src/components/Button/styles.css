.button-container {
  display: flex;
  justify-content: flex-start;
  width: 40vw; }

.button {
  height: 40px;
  width: 350px;
  border: solid 2px white;
  border-radius: 40px;
  display: flex;
  justify-content: center;
  align-items: center; }

.button:hover {
  cursor: pointer; }

.selected {
  color: #f7d700;
  border-color: #f7d700; }
