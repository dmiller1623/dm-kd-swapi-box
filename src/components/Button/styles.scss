.button-container {
  display: flex;
  justify-content: flex-start;
  width: 40vw;
}

.button {
  align-items: center;
  border: solid 2px white;
  border-radius: 40px;
  display: flex;
  height: 40px;
  justify-content: center;
  width: 350px;
}

.button:hover {
  cursor: pointer;
}

.selected {
  border-color: #f7d700;
  color: #f7d700;
}