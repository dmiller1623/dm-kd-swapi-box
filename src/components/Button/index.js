import React from 'react';
import './styles.css'
import PropTypes from 'prop-types'

const Button = ({ handleDisplayCards  }) => {
  return (
    <div className='button-container'>
      <div className='button people-button' onClick={(e) => handleDisplayCards(e)}>people</div>
      <div className='button planets-button'  onClick={(e) => handleDisplayCards(e)}>planets</div>
      <div className='button vehicles-button'  onClick={(e) => handleDisplayCards(e)}>vehicles</div>
    </div>
  )
}

Button.propTypes = {
  handleDisplayCards: PropTypes.func
}

export default Button;