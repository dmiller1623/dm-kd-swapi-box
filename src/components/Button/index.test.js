import React from 'react'
import Button from './index.js'
import { shallow } from 'enzyme'

describe('Button', () => {
  let wrapper
  let mockHandleDisplayCards

  beforeEach(() => {
    mockHandleDisplayCards = jest.fn()
    wrapper = shallow(<Button handleDisplayCards={mockHandleDisplayCards}/>)
  })
  
  it('should match the snapshot', () => {
    let wrapper = shallow(<Button />)
    expect(wrapper).toMatchSnapshot()
  })

  it('on click it should call handleDisplayCards', () => {
    wrapper.find('.people-button').simulate('click')
    expect(mockHandleDisplayCards).toHaveBeenCalled()
  })

  it('on click it should call handleDisplayCards', () => {
    wrapper.find('.planets-button').simulate('click')
    expect(mockHandleDisplayCards).toHaveBeenCalled()
  })

  it('on click it should call handleDisplayCards', () => {
    wrapper.find('.vehicles-button').simulate('click')
    expect(mockHandleDisplayCards).toHaveBeenCalled()
  })
})