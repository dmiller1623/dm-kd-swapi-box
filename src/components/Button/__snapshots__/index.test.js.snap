// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="button people-button"
          onClick={[Function]}
>
          people
</div>,
        <div
          className="button planets-button"
          onClick={[Function]}
>
          planets
</div>,
        <div
          className="button vehicles-button"
          onClick={[Function]}
>
          vehicles
</div>,
      ],
      "className": "button-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "people",
          "className": "button people-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "people",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "planets",
          "className": "button planets-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "planets",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "vehicles",
          "className": "button vehicles-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "vehicles",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="button people-button"
            onClick={[Function]}
>
            people
</div>,
          <div
            className="button planets-button"
            onClick={[Function]}
>
            planets
</div>,
          <div
            className="button vehicles-button"
            onClick={[Function]}
>
            vehicles
</div>,
        ],
        "className": "button-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "people",
            "className": "button people-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "people",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "planets",
            "className": "button planets-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "planets",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "vehicles",
            "className": "button vehicles-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "vehicles",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
